source/PlayerInfo.cpp

```diff
@@ -910,26 +915,33 @@ void PlayerInfo::BuyShip(const Ship *model, const string &name, bool isGift)
 			for(const auto &it : model->Outfits())
 				stock[it.first] -= it.second;
 		}
+		// Store named, unamed ships under their name or a generic name of their model so they can be checked later.
+		else
```

Comment:
It makes more sense to flip the condition so this is done in the `if` body, rather than the `else` body.

(I think you meant "unnamed" rather than unamed, too.)